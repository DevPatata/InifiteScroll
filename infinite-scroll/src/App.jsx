import { AppRouter } from './routes/AppRouter'
import { ItemsContextProvider } from './context/ItemsProvider'
import './App.css'

export const App = () => {
  return (
    <ItemsContextProvider>
      <AppRouter />
    </ItemsContextProvider>
  )
}
