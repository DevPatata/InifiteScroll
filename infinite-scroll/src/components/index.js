export { ListOfItems } from './listOfItems/listOfItems';
export { CardItems } from './cardItem/cardItem';
export { ButtonItem } from './button/button';
export { InputSearch } from './inputSearch/inputSearch';
export { ModalProduct } from './modalProduct/modalProduct';
export { Spinner } from './spinner/spinner';