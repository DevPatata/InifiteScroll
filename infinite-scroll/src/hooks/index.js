export { useItems } from './useItem';
export { useNearScreen } from './useNearScreen';
export { useForm } from './useForm';
export { useModal } from './useModal';