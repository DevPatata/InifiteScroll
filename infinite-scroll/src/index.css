:root {
  --color-grey: rgb(219 215 215);
  --color-light-grey: #f9f9f9;
  --color-font: #213547;
  font-family: Inter, system-ui, Avenir, Helvetica, Arial, sans-serif;
  line-height: 1.5;
  font-weight: 400;
  color-scheme: light dark;
  color: var(--color-font);
  background-color: rgb(249 248 248);
  font-synthesis: none;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.card{
  padding: 1rem;
}

  
