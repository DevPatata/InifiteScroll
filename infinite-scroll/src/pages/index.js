export { InfiniteScrollPage } from './infiniteScrollPages/infiniteScrollPage'