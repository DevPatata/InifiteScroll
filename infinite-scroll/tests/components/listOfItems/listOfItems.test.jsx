import React from 'react'
import { render, screen } from '@testing-library/react';
import { ListOfItems } from '../../../src/components';

describe('Prueba en componente ListOfItems', () => {
  test('Prueba match snaptshot', () => {
    const listItems = [];
    const { container } = render(<ListOfItems products={listItems} />);

    expect(container).toMatchSnapshot();
  });

  test('Prueba renderiza elementos pasados por parametro', () => {
    const listItems = [
      {
        img: 'urlImgTest',
        title: 'TitleTestDiferente',
        description: 'DescriptionTest',
        detail: { price: 0 }
      },
      {
        img: 'urlImgTest',
        title: 'TitleTest',
        description: 'DescriptionTest',
        detail: { price: 0 }
      }
    ];

    render(<ListOfItems products={listItems} />);

    expect(screen.getAllByRole('img').length).toBe(2);
    expect(screen.getByText('TitleTest')).toBeTruthy();
    expect(screen.getByText('TitleTestDiferente')).toBeTruthy();

  })
})