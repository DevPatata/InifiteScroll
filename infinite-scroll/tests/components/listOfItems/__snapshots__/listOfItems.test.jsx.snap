// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en componente ListOfItems Prueba match snaptshot 1`] = `
<div>
  <div
    class="list"
  />
  <article
    class="modal-close false"
    data-testid="modal-test-id"
  >
    <div
      class="background"
    />
    <div
      class="modal-position modal-container"
    >
      <span
        class="span-close"
      >
        <i
          aria-hidden="true"
          class="fa fa-times"
        />
      </span>
      <h1 />
      <h4 />
      <h6
        class="modal-address"
      >
        Address: 
      </h6>
    </div>
  </article>
</div>
`;
